### PR TITLE
Private/deliz/ca 331398

### PR DESCRIPTION
**Pull request by DeliZhangX**

### CODE REVIEW

**Comment by DeliZhangX** [comment]
All tests passed
- XenRT jobs: 2868528, 2868530, 2868641 and 2868756
- Manual test (see comments of CA-331398 to get details)

File: src/XenCert/XenCertCommon.py [hunk 1]
```diff
@@ -284,6 +284,12 @@ def getConfigWithHiddenPassword(config, storage_type):
         pass
     return config_with_hidden_password
 
+def hidePathInfoPassword(pathInfo, delimiter=':', password_index=2):
```
[the remark this answers is not in the document]
Comment (XihuanYang) [reply src/XenCert/XenCertCommon.py:287]: Could we test a hardware that not need a password in pathInfo?

File: src/XenCert/StorageHandler.py [hunk 2]
```diff
@@ -3413,15 +3417,19 @@ def Create(self):
 
             device_config = copy.deepcopy(self.device_config)
 
+            if 'targetIQN' in device_config and len(device_config['targetIQN'].split(',')) > 1:
+                device_config['targetIQN'] = '*'
```
Comment (minli1) [inline src/XenCert/StorageHandler.py:3421]: What is the purpose for setting the `targetIQN` tp `*`?

File: src/XenCert/StorageHandler.py [hunk 3]
```diff
@@ -3413,15 +3417,19 @@ def Create(self):
...
             # try to create an SR with one of the LUNs mapped, if all fails
             # throw an exception
             for scsiId in listSCSIId:
                 try:
                     device_config['SCSIid'] = scsiId
+                    device_config_tmp = getConfigWithHiddenPassword(device_config, self.storage_conf['storage_type'])
                     XenCertPrint("The SR create parameters are {}, {}".format(
                         util.get_localhost_uuid(self.session),
-                        device_config))
+                        device_config_tmp))
```
Comment (XihuanYang) [inline src/XenCert/StorageHandler.py:3432]: maybe device_config_print is more specific name for this value

File: src/XenCert/StorageHandler.py (hunk 2 above)
Comment (DeliZhangX) [reply src/XenCert/StorageHandler.py:3421]: It's copy from LVM handling, it's SM interface.

File: src/XenCert/XenCertCommon.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (DeliZhangX) [reply src/XenCert/XenCertCommon.py:287]: All support FC switches need password to login and password_index is always 2  so far, pathinfo is defined by us, not specific device.

File: src/XenCert/StorageHandler.py (hunk 3 above)
Comment (DeliZhangX) [reply src/XenCert/StorageHandler.py:3432]: It's copy from existing code, you can search from source code.

**Comment by DeliZhangX** [comment]
Thanks all!

**Pull request merged**